x/staking/keeper/val_state_change.go

```diff
@@ -341,6 +348,12 @@ func (k Keeper) beginUnbondingValidator(ctx sdk.Context, validator types.Validat
 		return validator, err
 	}
 
+	k.SetValidatorByUnbondingID(ctx, validator, id)
+
+	if err := k.Hooks().AfterUnbondingInitiated(ctx, id); err != nil {
```

Comment:
What is the semantic difference between `AfterUnbondingInitiated` and `AfterValidatorBeginUnbonding` hooks? What would you call in one but not the other?

Reply:
Good question. The difference is that `AfterUnbondingIntiated` is called for all unbonding operations. It is part of the standardized interface we added with our two hooks to allow unbondings to be paused and resumed, by Interchain Security and other modules. 

I'm not sure what uses `AfterValidatorBeginUnbonding`. It doesn't appear that anything in the SDK itself does. In any case, it is only for validators, and there are no equivalents for other types of unbondings. I see 4 options:

1. Leave it as-is and ship it. We have a weird situation in this function where two hooks have similar uses, but there is no harm done.
2. Make interchain security (and other modules who will use these hooks, maybe mesh security?) have a special case where for validator unbondings, but not for undelegations and redelegations, they use the `AfterValidatorBeginUnbonding` hook. IMO this makes the interface messier.
3. Break `AfterUnbondingIntiated` into 3 separate hooks, just for aesthetic reasons, e.g. `AfterValidatorBeginUnbonding`, `AfterDelegationBeginUnbonding`, `AfterUndelegationBeginUnbonding`. IMO this is overengineering because there is no real need to do it.
4. Delete `AfterValidatorBeginUnbonding`. Is anyone even using it?

I would opt for option 1.

Reply:
I guess we can leave this as-is. Doesn't seem a non-no-op `AfterValidatorBeginUnbonding` is implemented anywhere. Maybe we just delete it?